nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java

```diff
@@ -114,10 +121,13 @@ KafkaPublisherResult publish(PublishingContext publishingContext) {
 
         int prevLastAckedMessageIndex = publishingContext.getLastAckedMessageIndex();
         List<Future<RecordMetadata>> resultFutures = new ArrayList<>();
+        System.out.println(1 - prevLastAckedMessageIndex);
```

Comment:
Yep, i'll wait till you all done and will address this and whatever others in one shot